src/cc/arduino/plugins/wifi101/UpdaterImpl.java

```diff
@@ -50,42 +64,69 @@ public UpdaterImpl() throws Exception {
 		});
 		Base.setIcon(this);
 
-		for (WINC1500Firmware firmware : WINC1500Firmware.available)
-			getFirmwareSelector().addItem(firmware);
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.5.4", "firmwares/WINC1500/19.5.4/m2m_aio_3a0.bin", true, new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.5.2", "firmwares/WINC1500/19.5.2/m2m_aio_3a0.bin", true, new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.4.4", "firmwares/WINC1500/19.4.4/m2m_aio_3a0.bin", true, new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model A", "19.4.4", "firmwares/WINC1500/19.4.4/m2m_aio_2b0.bin", true, new ArrayList<String>(asList("Arduino/Genuino MKR1000", "Wifi 101"))));
```

Comment:
Arduino/Genuino MKR1000 can't use the `m2m_aio_2b0.bin` file, WiFi won't work since the file is for module A which is only on the WiFi101 shield.